cedar-policy-validator/src/extensions/datetime.rs

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Cedar Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+//! Note on panic safety
+//! If any of the panics in this file are triggered, that means that this file has become
+//! out-of-date with the decimal extension definition in Core.
+//! This is tested by the `extension_schema_correctness()` test
+
+use crate::extension_schema::{ArgumentCheckFn, ExtensionFunctionType, ExtensionSchema};
+use crate::types::{self, Type};
+use cedar_policy_core::ast::{Expr, ExprKind, Literal, Name};
+use cedar_policy_core::extensions::datetime;
+use itertools::Itertools;
+
+use super::eval_extension_constructor;
+
+/// Note on safety:
+/// This module depends on the Cedar parser only constructing AST with valid extension calls
+/// If any of the panics in this file are triggered, that means that this file has become
+/// out-of-date with the datetime extension definition in Core.
+
+// PANIC SAFETY see `Note on safety` above
+#[allow(clippy::panic)]
+fn get_argument_types(fname: &Name, datetime_ty: &Type, duration_ty: &Type) -> Vec<types::Type> {
+    if !fname.as_ref().is_unqualified() {
+        panic!("unexpected datetime extension function name: {fname}")
+    }
+    match fname.basename().as_ref() {
+        "datetime" | "duration" => vec![Type::primitive_string()],
+        "offset" => vec![datetime_ty.clone(), duration_ty.clone()],
+        "durationSince" => vec![datetime_ty.clone(), datetime_ty.clone()],
+        "toDate" => vec![datetime_ty.clone()],
```

Comment:
I don't understand why we're cloning types here... Let me take a closer look later. Feel free to ignore this comment in the meantime.